### DISCUSSION TITLE
Mapping offset assistants

### DISCUSSION THREAD

**Discussion by Akarinnnnn** [ppy/osu#14879] [Ideas]
Nowadays, beatmap editor has integrated a waveform graph. When you check those existed(especially those ranked earlier) maps you can find a gap between note and the corresponding peak volume, like this screenshot.
![image](https://user-images.githubusercontent.com/43724908/135105780-f9e791e8-b77f-4550-b7c2-e524c9a4c779.png)

One of the reasons is hitsound itself have a similar gap. In this case, we can just cut the hitsound to fit it, and this solution is recommended in ranking criteria. Another solution is move red lines earlier to fit this hitsound, I think this is probably the solution which the map used.

A more common factor I think, is mapper's audio offset setting and, "convention". Assume a mapper only clicks `T` to timing, the timing result will be affected by it's offset setting, it's personal preferences  it's machine, and existed maps("convention"). Earlier map has a gap, and the new map will probably have a similar gap.

**Comment by peppy** [comment]
I read this and have no idea what I read.

**Reply by Akarinnnnn** [reply]
May I use my mother tongue?
